packages/nx/src/native/tui/app.rs

```diff
@@ -774,17 +939,442 @@ impl App {
+                    {
+                        Focus::MultipleOutput(prev_pane)
+                    } else if !self.task_list_hidden {
+                        // Go to task list if it's visible
+                        Focus::TaskList
+                    } else {
+                        // If task list is hidden, wrap around to rightmost pane
+                        if let Some(last_pane) =
+                            (0..2).rev().find(|&idx| self.pane_tasks[idx].is_some())
+                        {
+                            Focus::MultipleOutput(last_pane)
+                        } else {
+                            // Shouldn't happen (would mean no panes)
+                            return;
+                        }
+                    }
+                } else {
+                    // We're at leftmost pane (index 0)
+                    if !self.task_list_hidden {
+                        // Go to task list if it's visible
+                        Focus::TaskList
+                    } else if num_panes > 1 {
+                        // If task list hidden and multiple panes, wrap to rightmost pane
+                        if let Some(last_pane) =
+                            (1..2).rev().find(|&idx| self.pane_tasks[idx].is_some())
+                        {
+                            Focus::MultipleOutput(last_pane)
+                        } else {
+                            // Stay on current pane if can't find another one
+                            Focus::MultipleOutput(current_pane)
+                        }
+                    } else {
+                        // Only one pane and task list hidden, nowhere to go
+                        Focus::MultipleOutput(current_pane)
+                    }
+                }
+            }
+            Focus::HelpPopup => Focus::TaskList,
+            Focus::CountdownPopup => Focus::TaskList,
+        };
+
+        self.update_focus(focus);
+    }
+
+    fn toggle_task_list(&mut self) {
+        // If there are no visible panes, do nothing, otherwise the screen will be blank
+        if !self.has_visible_panes() {
+            return;
+        }
+        self.task_list_hidden = !self.task_list_hidden;
+        self.layout_manager
+            .set_task_list_visibility(if self.task_list_hidden {
+                TaskListVisibility::Hidden
+            } else {
+                TaskListVisibility::Visible
+            });
+        self.recalculate_layout_areas();
+        // Ensure the pty instances get resized appropriately (no debounce as this is based on an imperative user action)
+        let _ = self.handle_pty_resize();
+    }
+
+    fn cycle_layout_modes(&mut self) {
+        // TODO: add visual feedback about layout modes
+        self.layout_manager.cycle_layout_mode();
+        self.recalculate_layout_areas();
+        // Ensure the pty instances get resized appropriately (no debounce as this is based on an imperative user action)
```

Comment:
Can you explain when we need to debounce and when we don't to me one more time please?